testapp/exam/code_server.py

```diff
@@ -649,106 +363,131 @@ def _check_java_code(self, ref_code_path, submit_code_path):
                         err = err + "\n" + e
             except:
                 err = err + "\n" + stdnt_stderr
-        return success, err
 
-    def _remove_null_substitute_char(self, string):
-        """Returns a string without any null and substitute characters"""
-        stripped = ""
-        for c in string:
-            if ord(c) is not 26 and ord(c) is not 0:
-                stripped = stripped + c
-        return ''.join(stripped)
- 
-    def run_scilab_code(self, answer, test_code, in_dir=None):
-        """Tests given Scilab function (`answer`) with the `test_code`
-        supplied. If the optional `in_dir` keyword argument is supplied
-        it changes the directory to that directory (it does not change
-        it back to the original when done). This function also timesout
-        when the function takes more than SERVER_TIMEOUT seconds to run
-        to prevent runaway code.
-
-        The testcode is a path to the reference code.
-        The reference code will call the function submitted by the student.
-        The reference code will check for the expected output.
+        return success, err
 
-        If the path's start with a "/" then we assume they are absolute paths.
-        If not, we assume they are relative paths w.r.t. the location of this
-        code_server script.
+    def check_bash_script(self, ref_path, submit_path,
```

Comment:
See this should be in a separate class and not muddle the basic `TestCode` class.